### PR TITLE
Image-url-helper refactor

### PR DESCRIPTION
**Pull request by Halamix2**
**Description**

Changes proposed in this pull request:

- move parts used by all subcommands into its own `common` package

**Related issue(s)**

### CODE REVIEW

File: development/image-url-helper/pkg/common/images.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	inconsistent := make(ImageMap)
+
+	for imageName, image := range images {
+		for image2Name, image2 := range images {
+			if image.ImageURL() == image2.ImageURL() && imageName != image2Name {
+				inconsistent[imageName] = image
+				inconsistent[image2Name] = image2
+			}
+		}
+	}
```
Comment (dekiel) [inline development/image-url-helper/pkg/common/images.go:47]: This should work as well but with only one iteration over a ImageMap.
```suggestion
        tmpImages[string]string
	for imageName, image := range images {
	        if _, ok := tmpImages[image.ImageURL()]; ok {
	                if imageName !=  tmpImages[image.ImageURL()] {
				inconsistent[imageName] = image
			} else {
			        tmpImages[image.ImageURL()] = imageName
			}
		}
	}
```

File: development/image-url-helper/pkg/common/valuefile.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+func AppendImagesToMap(parsedFile ValueFile, images, testImages ImageMap, component string, components ImageToComponents) {
+	for _, image := range parsedFile.Global.Images {
+		// add registry info directly into the image struct
+		if image.ContainerRegistryPath == "" {
+			image.ContainerRegistryPath = parsedFile.Global.ContainerRegistry.Path
+		}
+		images[image.FullImageURL()] = image
+
+		components[image.FullImageURL()] = append(components[image.FullImageURL()], component)
```
Comment (dekiel) [inline development/image-url-helper/pkg/common/valuefile.go:28]: Why you need two map, where they are indexed by the same value? Can ImageMap type hold information about components where it is used?

File: development/image-url-helper/pkg/common/valuefile.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	for _, testImage := range parsedFile.Global.TestImages {
```
Comment (dekiel) [inline development/image-url-helper/pkg/common/valuefile.go:31]: Why you make a difference between testImages and images?

**Review by dekiel — CHANGES_REQUESTED** [review]

File: development/image-url-helper/pkg/common/valuefile.go (hunk 3 above)
Comment (Halamix2) [reply development/image-url-helper/pkg/common/valuefile.go:40]: `List` subcommand have option to skip testImages

File: development/image-url-helper/pkg/common/valuefile.go [hunk 4]
```diff
@@ -17,22 +17,44 @@ type ValueFile struct {
...
 		if image.ContainerRegistryURL == "" {
 			image.ContainerRegistryURL = parsedFile.Global.ContainerRegistry.Path
 		}
-		images[image.FullImageURL()] = image
 
-		components[image.FullImageURL()] = append(components[image.FullImageURL()], component)
+		if _, ok := images[image.FullImageURL()]; ok {
+			componentImage := images[image.FullImageURL()]
+			componentImage.Components[component] = true
+			images[image.FullImageURL()] = componentImage
```
Comment (dekiel) [inline development/image-url-helper/pkg/common/valuefile.go:30]: Can you work directly on images `ComponentImageMap`?
```suggestion
                        images[image.FullImageURL()].Components[component] = true
```

File: development/image-url-helper/pkg/common/valuefile.go [hunk 5]
```diff
@@ -17,22 +17,44 @@ type ValueFile struct {
...
+		} else {
+			componentImage := ComponentImage{} //Image: image, Components: make(map[string]bool)}
+			componentImage.Image = image
+			componentImage.Components = make(map[string]bool)
+			componentImage.Components[component] = true
+			images[image.FullImageURL()] = componentImage
```
Comment (dekiel) [inline development/image-url-helper/pkg/common/valuefile.go:36]:
```suggestion
images[image.FullImageURL()] = ComponentImage{
		Components: map[string]bool{component: true},
		Image:      image,
	}
```
Can you assign `ComponentImage` literal. It's easier to understand what you are doing here than with multiple assignment.

**Review by dekiel — CHANGES_REQUESTED** [review]

File: development/image-url-helper/pkg/common/images.go [hunk 6]
```diff
@@ -36,29 +42,42 @@ func (i Image) ImageURL() string {
...
+	for key, _ := range source {
 		if _, ok := target[key]; !ok {
-			target[key] = val
+			// we have the same image in both maps
+			// TODO
+			tmpComponentImage := ComponentImage{Image: source[key].Image}
+
+			if tmpComponentImage.Components == nil {
+				tmpComponentImage.Components = make(map[string]bool)
+			}
```
[the remark this answers is not in the document]
Comment (Halamix2) [reply development/image-url-helper/pkg/common/images.go:73]: if `tmpComponentImage.Components` is nil I cannot `append()` to it unless I initialize it here

**Review by dekiel — APPROVED** [review]

**Pull request merged**